### PR TITLE
Fix: use of React.PropTypes

### PR DESCRIPTION
**Pull request by sonhanguyen**
##### Changes
- Remove use of React.PropTypes
- Remove some react warnings in tests that are not critical but annoying

### CODE REVIEW

File: src/components/adslot-ui/HoverDropdownMenu/index.spec.jsx [hunk 1]
```diff
@@ -62,21 +62,21 @@ describe('HoverDropdownMenuComponent', () => {
...
-        {_.map(links, (link) => <HoverDropdownMenu.Item {...link} />)}
+        {_.map(links, (link, idx) => <HoverDropdownMenu.Item key={idx} {...link} />)}
       </HoverDropdownMenu>
     );
-    component.find('.hover-dropdown').simulate('mouseEnter', { target: 'some target' });
+    component.find('.hover-dropdown').simulate('mouseEnter', { target: _.noop });
     expect(component.state()).to.eql({
       isOpen: true,
-      target: 'some target',
+      target: _.noop,
```
Comment (sonhanguyen) [inline src/components/adslot-ui/HoverDropdownMenu/index.spec.jsx]: the proptype does not accept string

**Comment by omgaz** [comment]
LGTM but should target be defined as something else like `<div></div>`

File: src/components/adslot-ui/HoverDropdownMenu/index.spec.jsx (hunk 1 above)
Comment (sharmajyo) [reply src/components/adslot-ui/HoverDropdownMenu/index.spec.jsx]: you mean, we can't assign any string value to prop? or its just some special case where we can't

Comment (sonhanguyen) [reply src/components/adslot-ui/HoverDropdownMenu/index.spec.jsx]: srr I mean `Overlay.target` does not accept string, it does accept `function(): DomElement` though

File: src/components/adslot-ui/HoverDropdownMenu/index.spec.jsx [hunk 2]
```diff
@@ -139,6 +140,7 @@ describe('HoverDropdownMenuComponent', () => {
 
     beforeEach(() => {
       popoverProps = {
+        id: 'popover',
```
Comment (sonhanguyen) [inline src/components/adslot-ui/HoverDropdownMenu/index.spec.jsx]: required for accessibility

File: src/components/adslot-ui/ListPicker/index.jsx [hunk 3]
```diff
@@ -161,14 +161,14 @@ class ListPickerComponent extends React.Component {
           {
             _.isEmpty(linkButtons) ? null :
             <div className="pull-left">
-              {_.map(linkButtons, (linkButton) => (
+              {_.map(linkButtons, (linkButton, key) => (
```
Comment (xiaofan2406) [inline src/components/adslot-ui/ListPicker/index.jsx]: just curious why this change?

File: src/components/adslot-ui/HoverDropdownMenu/PopoverLinkItem/index.jsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 /* eslint-disable react/prop-types */
```
Comment (xiaofan2406) [inline src/components/adslot-ui/HoverDropdownMenu/PopoverLinkItem/index.jsx]: is this line relevant to remove?

File: src/components/adslot-ui/ListPicker/index.jsx (hunk 3 above)
Comment (sonhanguyen) [reply src/components/adslot-ui/ListPicker/index.jsx]: https://github.com/Adslot/adslot-ui/pull/656/files#diff-8e90d83db60d7e3897dc4457f7ea133eR171

File: src/components/adslot-ui/HoverDropdownMenu/PopoverLinkItem/index.jsx (hunk 4 above)
Comment (sonhanguyen) [reply src/components/adslot-ui/HoverDropdownMenu/PopoverLinkItem/index.jsx]: that looks like it's for some `renderX` function to pass linter, not really causes any runtime warning atm

Comment (sonhanguyen) [reply src/components/adslot-ui/HoverDropdownMenu/PopoverLinkItem/index.jsx]: srr it's actually for [line #42](https://github.com/Adslot/adslot-ui/pull/656/files/20055cfd494c96851feeff79b360dd6354e0da33#diff-0b124245e4f0bfd4f3073eacb42bd13dR42) to deal with eslint not being able to analyse

Comment (vinteo) [reply src/components/adslot-ui/HoverDropdownMenu/PopoverLinkItem/index.jsx]: is it possible to move it to disable for just that line?

**Comment by lightbringer1991** [comment]
merging ...
